packages/rum-core/src/domain/limitModification.ts

```diff
@@ -14,43 +15,85 @@ export function limitModification<T extends Context, Result>(
 ): Result | undefined {
   const clone = deepClone(object)
   const result = modifier(clone)
+
+  // Validate the modified fields and assign them back to 'original' if they match the expected type.
   objectEntries(modifiableFieldPaths).forEach(([fieldPath, fieldType]) => {
-    const newValue = get(clone, fieldPath)
-    const newType = getType(newValue)
-    if (newType === fieldType) {
-      set(object, fieldPath, sanitize(newValue))
+    const pathSegments = fieldPath.split('.')
```

Comment:
I love the approach! The one suggestion I have is to tweak the implementation strategy slightly. If you change this line to:
```suggestion
    const pathSegments = fieldPath.split(/\.|(?=\[\])/)
```
Then instead of `foo.bar[].baz` being split into `['foo', 'bar[]', 'baz']`, it will be split into `['foo', 'bar', '[]', 'baz']`. This should lead to a cleaner implementation of `getValueFromPath()`, where you can treat `[]` as a special field name that maps over all elements of an array. With the appropriate changes to `getValueFromPath()`, it will also mean that `foo.bar[][].baz` will work as expected with no special effort, so that we can support modifications of nested arrays.